site/en/docs/extensions/mv3/storage-and-cookies/index.md

```diff
@@ -0,0 +1,107 @@
+some cases these behave differently in extensions.
+
+This article does not cover extension specific APIs. For more on these, see
+[chrome.storage][chrome-storage-api].
+
+## Storage {: #storage }
+
+### Persistence {: #storage-persistence }
+
+Extension storage is not cleared when a user uses the "Clear Browsing Data" feature. This includes
+any data stored using web storage APIs (such as Local Storage and IndexedDB).
+
+By default, extensions are subject to the normal quota restrictions on storage, which can be checked
+using the `navigator.storage.estimate()` method. Storage can also be evicted under heavy memory
+pressure, although this is rare. To avoid this:
+
+- Request the `unlimitedStorage` permission, which affects both extension and web storage APIs and
+exempts extensions from both quota restrictions and eviction.
+- Use [`navigator.storage.persist()`][storage-persist] for protection against eviction.
+
+Extension storage is shared across the extension's origin, such as the associated service worker,
+any extension pages including popups and the side panel, and offscreen documents. In content
+scripts, calling web storage APIs accesses data from the origin the content script is injected on
+and not the extension.
+
+### Access in service workers {: #storage-in-service-workers }
+
+Some web storage APIs like IndexedDB are accessible in service workers. However, Local Storage and
+Session Storage are not.
+
+Instead, use an [offscreen document][offscreen]. For example, you could use the following to migrate
+data when updating from Manifest V2 to Manifest V3:
+
+1. In the extension service worker check `chrome.storage` for your data.
+1. If your data isn't found, [create][create-offscreen] an offscreen document and call
+[`sendMessage()`][send-message] to start a conversion routine. Read the data from the web storage
+APIs and use message passing to return it to the service worker.
+1. In the service worker, write the data to a new location.
+
+### Partitioning {: #storage-partitioning }
+
+To prevent certain types of cross-site tracking, [storage partitioning][storage-partitioning]
+introduces changes to how partitioning keys are defined. In practice, this means that if site A
+embeds an iframe containing site B, site B will not be able to access the same storage it would
+usually have when navigated to directly.
+
+To mitigate the impact of this in extensions, two key exemptions apply:
+
+- If a page with the `chrome-extension://` scheme is embedded in any site, storage partitioning will
+not apply, and the extension will have access to its top-level partition.
+- If a page with the `chrome-extension://` scheme includes an iframe, and the extension has
+[host permissions][declare-permissions] for the site it is embedding, that site will also have
+access to its top-level partition.
+
+Combined, these mean that storage partitioning is not usually something which extension developers
```

Comment:
"Not usually" makes me weary as I'm instantly wondering which cases are not usual...

Reply:
I ended up removing this line. Given this isn't a migration guide, "test your extension with these changes" didn't make sense, and I don't think there is much else to say.